ui/index.jsx

```diff
@@ -277,8 +277,16 @@ function AppWrapper() {
     // @endif
   }, []);
 
+  useEffect(() => {
+    if (persistDone) {
+      app.store.dispatch(doToggle3PAnalytics());
+    }
+  }, [persistDone]);
+
   useEffect(() => {
     if (readyToLaunch && persistDone) {
+      analytics.startupEvent();
```

Comment:
This shouldn't be moved. We still want to start it immediately, so the time to start is still accurate. It should delay calling the `readyEvent`

Reply:
Hm, this is code that runs before we have the persisted GA preference value from store. Maybe need another earlier place to persist it ?

Reply:
We don't want to send GA events until we know we have permission.

Reply:
It can start counting before we know if users are sharing analytics or not. If they aren't, we can just discard that value

Reply:
startupEvent() does the actual sendGA()
startTime is still set at the top of the file.

readyEvent() includes the total start duration.
const appReadyTime = Date.now();
const timeToStart = appReadyTime - startTime;
analytics.readyEvent(timeToStart);

Reply:
I ended up putting it back and trust the localStorage.